### Carregamento de bibliotecas

No bloco de código abaixo é feito o carregamento de todas as bibliotecas que serão utilizadas neste notebook.

In [10]:
from scipy.io import loadmat
import numpy as np

### Carregamento

No bloco de código abaixo, realizamos o carregamento de todos os pedaços da base de dados avi em `single-channel`. 

É efetuada uma iteração entre 1 e 5 e concatenados os dados carregados na variável `avi_data`.

In [11]:
# pré-carregamento
beta_data = loadmat("../../datasets/beta/S1.mat", struct_as_record=True)

### Captura dos Dados

No trecho de código abaixo é executada a captura dos dados (X) e suas respectivas classes (y). 

In [12]:
# extração direta dos dados

X = beta_data['data'][0][0][0]
y = beta_data['data'][0][0][1][0][0][4][0]

print(X.shape, "--> (ELETRODOS, DADOS, 4, TARGET)")

print("\nDivisão dos rótulos sem duplicatas (sequências):")
print(y.shape)
print(y)



(64, 750, 4, 40) --> (ELETRODOS, DADOS, 4, TARGET)

Divisão dos rótulos sem duplicatas (sequências):
(40,)
[ 8.6  8.8  9.   9.2  9.4  9.6  9.8 10.  10.2 10.4 10.6 10.8 11.  11.2
 11.4 11.6 11.8 12.  12.2 12.4 12.6 12.8 13.  13.2 13.4 13.6 13.8 14.
 14.2 14.4 14.6 14.8 15.  15.2 15.4 15.6 15.8  8.   8.2  8.4]


### Organização dos dados

No trecho de código abaixo é realizado o `reshape` dos dados presentes na variável `X`.

O algoritmo realizará a seguinte transformação:

De

- (64, 750, 4, 40)

Para

- (4, 40, 64, 750)

In [13]:
print(X.shape)
reshape_X = np.array(np.transpose(X,  (2, 3, 0, 1)))
print(reshape_X.shape)
reshape_X = reshape_X.reshape(reshape_X.shape[0] * reshape_X.shape[1], reshape_X.shape[2], reshape_X.shape[3])
reshape_X.shape

(64, 750, 4, 40)
(4, 40, 64, 750)


(160, 64, 750)

### Organização dos dados

No trecho de código abaixo é realizado o `reshape` dos dados presentes na variável `y`.

O algoritmo realizará a seguinte transformação:

De

- (10, 10)

Para

- (100)

### Salvando dados em um arquivo

No trecho de código abaixo é feito o salvamento dos dados formatados na extensão `.npy`.

In [14]:
reshape_Y = np.array([np.transpose(data) for data in y])
print(reshape_Y)

[ 8.6  8.8  9.   9.2  9.4  9.6  9.8 10.  10.2 10.4 10.6 10.8 11.  11.2
 11.4 11.6 11.8 12.  12.2 12.4 12.6 12.8 13.  13.2 13.4 13.6 13.8 14.
 14.2 14.4 14.6 14.8 15.  15.2 15.4 15.6 15.8  8.   8.2  8.4]


In [15]:
# salva os dados numpy

np.save("../../datasets/beta/data.npy", reshape_X)
np.save("../../datasets/beta/labels.npy", reshape_Y)

### Tarefas:

- Padronizar os dados de todos os 4 participantes dos dados *single-target* e salvar de forma adequada para utilização nos próximos notebooks.
- Realizar o mesmo estudo para os dados *multi-target*